lib/rspec/core.rb

```diff
@@ -56,6 +56,7 @@ class << self
   # they use the runner multiple times within the same process. Users must deal
   # themselves with re-configuration of RSpec before run.
   def self.reset
+    @world.example_groups.reduce(&:remove_all_constants)
```

Comment:
`reduce` allows you to apply a calculation to an entire collection to create single "reduced" value out of the input list.  Here you aren't using it for that, though; you're just using it as a way to apply a side effect to every element in the collection.  Pretty much any of the `Enumerable` methods would work just as well here (e.g. `map`, `select`, `reject`, etc), but the one that is specifically intended to apply a side effect to every element in a collection is `each`.  Please use that instead as it communicates intent better:

``` ruby
@world.example_groups.each(&:remove_all_constants)
```